pkg/handler/collector/oci/oci.go

```diff
@@ -54,84 +54,87 @@ func NewOCICollector(ctx context.Context, repoTags map[string][]string, poll boo
 
 // RetrieveArtifacts get the artifacts from the collector source based on polling or one time
 func (o *ociCollector) RetrieveArtifacts(ctx context.Context, docChannel chan<- *processor.Document) error {
-	if o.poll {
-		for {
-			for repo, tags := range o.repoTags {
-				// when polling if tags are specified, it will never get any new tags
-				// that might be added after the fact. Defeating the point of the polling
-				if len(tags) > 0 {
-					return errors.New("image tag should not specified when using polling")
-				}
-				err := o.getTagsAndFetch(ctx, repo, tags, docChannel)
-				if err != nil {
-					return err
-				}
-				// set interval to about 5 mins or more
-				time.Sleep(o.interval)
-			}
-		}
-	} else {
+	if docChannel == nil {
+		return fmt.Errorf("nil channel passed to %s", OCICollector)
+	}
+
+	for {
 		for repo, tags := range o.repoTags {
-			err := o.getTagsAndFetch(ctx, repo, tags, docChannel)
-			if err != nil {
-				return err
+			if len(tags) > 0 && o.poll {
+				return errors.New("image tag should not specified when using polling")
+			}
+			if err := o.getTagsAndFetch(ctx, repo, tags, docChannel); err != nil {
+				return fmt.Errorf("fetching artifacts for %s: %w", repo, err)
 			}
 		}
+		if !o.poll {
+			break
+		}
+		select {
```

Comment:
This should be added to the other collectors as well? Another PR but worth adding. Thanks.

Reply:
That will be another PR. Thanks